### PR TITLE
(guide): add custom provider as option for saml sso

### PR DESCRIPTION
**Pull request by alexisintech**
Updated overview to mention the three IdPs we currently directly support integrations for, and to mention the guide for configuring a custom IdP (convo for this change lives [here](https://clerkinc.slack.com/archives/CHZ1FBBEG/p1711964994313999?thread_ts=1711662244.196599&cid=CHZ1FBBEG)):
🔎 https://docs-preview-868.clerkpreview.com/docs/authentication/saml/overview

Added guide for configuring SAML SSO with a custom IdP in Clerk:
🔎 https://docs-preview-868.clerkpreview.com/docs/authentication/saml/custom-provider

### CODE REVIEW

File: docs/authentication/saml/custom-provider.mdx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+### Map your IdP's claims to Clerk fields
+
+Mapping the claims in your IdP to the attributes in Clerk ensures that the data from your IdP is correctly mapped to the data in Clerk.
+
+In the Clerk Dashboard, find the **Attribute mapping** section. Here, you are shown what properties on the [`User`](/docs/users/overview#user-object) object in Clerk are being mapped to the claims in your IdP.
+
+In your IdP dashboard, there should be a section where you can map the IdP's claims to the attributes in Clerk. For example, Google has a `Primary email` claim that needs to be mapped to Clerk's `mail` property. During SAML configuration in the Google dashboard, Google provides a section where these claims can be mapped.
+
+If you have additional claims that you would like to map to Clerk that are not listed in the **Attribute mapping** section, you can do so by following the steps in the [Map other claims](#map-other-claims-optional) section.
```
Comment (S3Prototype) [inline docs/authentication/saml/custom-provider.mdx:98]:
```suggestion
If you have additional claims that you would like to map to Clerk that are not listed in the **Attribute mapping** section, you can do so by following the steps in the [Map other claims](#map-every-other-claim-optional) section.
```

File: docs/authentication/saml/custom-provider.mdx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+### Map every other claim (optional)
+
+In Clerk, the [`User`](/docs/users/overview#user-object) object has a `publicMetadata` property that you can use to store additional information about your users.
+
+To map other claims from your IdP that do not have a direct mapping to Clerk attributes, you can map them to the `publicMetadata` property. To do this, prepend the Clerk claims with `public_metadata_` during the mapping process.
+
+For example, say you were using Google as your IdP. Google users have the "Phone number" attribute. Clerk does not have a direct mapping for this attribute, as you can see in the Clerk Dashboard in the **Attribute mapping** section. Instead, in the Google dashboard, you would map Google's "Phone number" claim to `public_metadata_phone_number`. Then, in Clerk, the value for the user's phone number would be saved in the user's `User.publicMetadata` under the key `phone_number`.
+
+[Learn more about how to access the metadata from our APIs.](https://clerk.com/docs/users/metadata)
```
Comment (S3Prototype) [inline docs/authentication/saml/custom-provider.mdx:108]:
```suggestion
[Learn more about how to access the metadata from our APIs.](/docs/users/metadata)
```

**Review by S3Prototype — APPROVED** [review]

File: docs/authentication/saml/custom-provider.mdx (hunk 1 above)
Comment (alexisintech) [reply docs/authentication/saml/custom-provider.mdx:98]: ahh, I failed to change the heading

File: docs/authentication/saml/custom-provider.mdx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+- **Entity ID** - This is your application's URL that your IdP will redirect your users back to after they have authenticated in your IdP.
+- **Metadata URL** - This is the URL to your SAML connection's metadata file. This is the recommended way to configure your service provider.
+
+To find the values for these settings:
+
+1. In the Clerk Dashboard, find the **Service Provider configuration** section.
+2. Copy the values you need for your IdP.
+3. In your IdP dashboard, paste the values in the appropriate fields.
+
+### Configure your identity provider
```
Comment (chanioxaris) [inline docs/authentication/saml/custom-provider.mdx:62]:
```suggestion
### Configure your Identity Provider
```

File: docs/authentication/saml/custom-provider.mdx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+Create a new application in your IdP. In the next steps, you will configure your IdP with the settings provided by your service provider (Clerk), and you will configure Clerk with the settings provided by your IdP. So, you will need to keep both the IdP and Clerk dashboards open.
+
+### Configure your service provider
+
+To configure your service provider (Clerk), your IdP will either ask for the **Assertion Consumer Service (ACS) URL** and **Entity ID** or it will ask for the **Metadata URL**. If your IdP gives you the option to choose between the two, it is recommended to choose the **Metadata URL** as it is the quickest and most reliable way to configure your service provider.
+
+Here are what these settings mean:
+
+- **Assertion Consumer Service (ACS) URL** - This is a unique identifier for your SAML connection that your IdP application needs.
+- **Entity ID** - This is your application's URL that your IdP will redirect your users back to after they have authenticated in your IdP.
```
Comment (chanioxaris) [inline docs/authentication/saml/custom-provider.mdx:53]: Need to reverse these two. 
ACS URL is the application url where the user will be redirected
Entity ID is the unique identifier

File: docs/authentication/saml/custom-provider.mdx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+1. In the Clerk Dashboard, find the **Service Provider configuration** section.
+2. Copy the values you need for your IdP.
+3. In your IdP dashboard, paste the values in the appropriate fields.
+
+### Configure your identity provider
+
+To configure your IdP in Clerk, scroll down to the **Identity Provider configuration** section. There are two options for configuring your IdP:
+
+- [**Metadata configuration**](#metadata-configuration) - This is where you can input the URL to your IdP's metadata file. This is the recommended way to configure your IdP, but not all IdPs support this method.
```
Comment (chanioxaris) [inline docs/authentication/saml/custom-provider.mdx:66]: There are two ways to configure the IdP using the metadata configuration. Either by a metadata url or by a metadata file. Usually n IdP offers one of the two ways

File: docs/authentication/saml/custom-provider.mdx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+2. In the Clerk Dashboard, find the **Identity Provider configuration** section.
+3. Under the **Metadata configuration** option, input the metadata URL or upload the metadata file that you got from your IdP.
+4. Select the **Save changes** button to complete the setup.
+
+#### Custom configuration
+
+If you choose to manually input the configuration settings for your IdP, you will need to fill these three fields in the Clerk Dashboard:
+
+- **SSO URL** - This is the unique identifier of your IdP application.
+- **Entity ID** - This is your IdP's URL that Clerk will redirect your users to so that they authenticate.
```
Comment (chanioxaris) [inline docs/authentication/saml/custom-provider.mdx:81]: Need to reverse these two as well. SSO URL is used to authenticate users on IdP and Entity ID is the unique identifier

**Review by JacobMGEvans — APPROVED** [review]
I see there are some pending comments, besides that it looks great!

File: docs/authentication/saml/custom-provider.mdx [hunk 7]
```diff
@@ -0,0 +1,123 @@
+---
+title: Custom provider
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply docs/authentication/saml/custom-provider.mdx:2]: Yep, whatever you feel like it's the best, no strong opinion on this!

**Pull request merged**